tsl/src/compression/algorithms/uuid_compress.c

```diff
@@ -14,30 +14,49 @@
 #include "lib/hyperloglog.h"
 #include "null.h"
 #include "simple8b_rle.h"
+#include "utils/palloc.h"
 
 #ifdef TS_USE_UMASH
 #include "import/umash.h"
 #endif
 
+typedef union UuidType
+{
+	uint64 components[2];
+	pg_uuid_t uuid;
+} UuidType;
+
+typedef enum
+{
+	UUID_COMPRESS_SUBTYPE_DELTADELTA = 0,
+	UUID_COMPRESS_SUBTYPE_COUNT = 1, /* Must be last */
+} UuidCompressSubtype;
```

Comment:
So you're also introducing algorithm versions with this PR? I imagined a simpler system with adding different algorithms, but this might also be OK.